modules/gapi/include/opencv2/gapi/ocl/goclkernel.hpp

```diff
@@ -149,12 +149,12 @@ struct tracked_cv_umat{
 struct scalar_wrapper_ocl
 {
     //FIXME reuse CPU (OpenCV) plugin code
-    scalar_wrapper_ocl(cv::gapi::own::Scalar& s) : m_s{cv::gapi::own::to_ocv(s)}, m_org_s(s) {};
+    scalar_wrapper_ocl(cv::Scalar& s) : m_s{s}, m_org_s(s) {};
     operator cv::Scalar& () { return m_s; }
-    void writeBack() const  { m_org_s = to_own(m_s); }
+    void writeBack() const  { m_org_s = m_s; }
 
     cv::Scalar m_s;
-    cv::gapi::own::Scalar& m_org_s;
+    cv::Scalar& m_org_s;
```

Comment:
Is it really necessary keep two `cv::Scalar` here ?

Reply:
Fixed. Same problem in gcpukernel.hpp.